src/status_im2/contexts/chat/bottom_sheet_composer/gesture.cljs

```diff
@@ -0,0 +1,103 @@
+      (when (and @expanding? (not max-opacity?))
+        (reanimated/set-shared-value opacity (Math/abs progress))))))
+
+(defn maximize
+  [{:keys [maximized?]}
+   {:keys [height saved-height background-y opacity]}
+   {:keys [max-height]}]
+  (reanimated/animate height max-height)
+  (reanimated/set-shared-value saved-height max-height)
+  (reanimated/set-shared-value background-y 0)
+  (reanimated/animate opacity 1)
+  (reset! maximized? true)
+  (rf/dispatch [:chat.ui/set-input-maximized true]))
+
+(defn minimize
+  [{:keys [input-ref emoji-kb-extra-height saved-emoji-kb-extra-height]}]
+  (when @emoji-kb-extra-height
+    (reset! saved-emoji-kb-extra-height @emoji-kb-extra-height)
+    (reset! emoji-kb-extra-height nil))
+  (rf/dispatch [:chat.ui/set-input-maximized false])
+  (.blur ^js @input-ref))
+
+(defn bounce-back
+  [{:keys [height saved-height opacity background-y]}
+   {:keys [window-height]}
+   starting-opacity]
+  (reanimated/animate height (reanimated/get-shared-value saved-height))
+  (when (zero? starting-opacity)
+    (reanimated/animate opacity 0)
+    (reanimated/animate-delay background-y (- window-height) 300)))
+
+(defn drag-gesture
+  [{:keys [input-ref] :as props}
+   {:keys [gesture-enabled?] :as state}
+   {:keys [height saved-height last-height opacity background-y container-opacity] :as animations}
+   {:keys [max-height lines] :as dimensions}
+   keyboard-shown]
+  (let [expanding?       (atom true)
+        starting-opacity (reanimated/get-shared-value opacity)]
+    (->
+      (gesture/gesture-pan)
+      (gesture/enabled @gesture-enabled?)
+      (gesture/on-start (fn [e]
+                          (if-not keyboard-shown
+                            (do ; focus and end
+                              (when (< (oops/oget e "velocityY") c/velocity-threshold)
+                                (reanimated/set-shared-value container-opacity 1)
+                                (reanimated/set-shared-value last-height max-height))
+                              (.focus ^js @input-ref)
+                              (reset! gesture-enabled? false))
+                            (do ; else, will start gesture
+                              (reanimated/set-shared-value background-y 0)
+                              (reset! expanding? (neg? (oops/oget e "velocityY")))))))
+      (gesture/on-update (fn [e]
+                           (let [translation (oops/oget e "translationY")
+                                 min-height  (utils/get-min-height lines)
+                                 new-height  (+ (- translation)
+                                                (reanimated/get-shared-value saved-height))
+                                 new-height  (utils/bounded-val new-height min-height max-height)]
+                             (when keyboard-shown
+                               (reanimated/set-shared-value height new-height)
+                               (set-opacity e
```

Comment:
I always wonder if it is better to pass atoms derefed or not? Is there any difference?

Reply:
Indeed, there's an important difference. If the atom *value* is passed, then the parent component was responsible for deref'ing the atom, hence if the value of the atom changes, the parent component will be re-rendered. If, instead, we pass the atom itself and deref inside the child component, the parent component won't be re-rendered (if the atom is not being deref'ed in other parts of the parent component of course).

Also, as you know, passing an atom (not its value) is sometimes desirable when we want to delay deref'ing the value, for example when we only want to deref inside an *on press* event.

Reply:
So from what I understand it is generally better to pass the atom itself, right?

Note that `drag-gesture` and `set-opacity` are not components.

Reply:
I think there's no definitive answer @OmarBasem, because it depends on what we want to do. I usually prefer to pass deref'ed values because the more pure functions in the code the simpler. But oftentimes passing the atom itself is the only way to achieve what's desired.

My rule of thumb is: can the same result be achieved by passing a value instead of an atom without impacting other non-functional requirements? If the answer is yes, then I pass a deref'ed atom (value), otherwise pass the atom.

Reply:
Sorry for getting into the discussion, just wanted to give my opinion.

I agree with @ilmotta, I think it's the same as avoid having global atoms, it helps to keep the functional programming style which IMO is simplier.

Related to pass atoms, I usually avoid it and only pass the deref-ed value, it's also easier for devs to create/modify the child component since you don't need to know what parameter is an atom and use `@` or the component crashes.

If I need to perform an update inside the child component, most of the time it's possible to pass a callback, e.g.
```clojure
;; Passing an atom:
(defn child-component [{:keys [value]}]
  ;; ... code
  (when something
    (reset! value "something else"))
  ;; ... code
  )

[rn/view
 [child-component {:value my-atom}]]
 
 ;; -----------------------------------------------
 
 ;; Passing a callback
 (defn child-component [{:keys [value update-value]}]
  ;; ... code
  (when something
    (update-value value))
  ;; ... code
  )

[rn/view
 [child-component {:value @my-atom
                   :update-value #(reset! my-atom "something else")}]]
```

I hope this example helps :)

Reply:
Thanks for the addition @ulisesmac